medicines/api/src/main.rs

```diff
@@ -15,88 +16,62 @@ mod substance;
-            http::header::ACCEPT,
-            http::header::CONTENT_TYPE,
-        ])
-        .max_age(3600)
-        .finish()
+pub fn healthz() -> impl Filter<Extract = impl Reply, Error = Rejection> + Copy {
+    warp::path!("healthz")
+        .and(warp::get())
+        .map(warp::reply)
+        .map(|reply| warp::reply::with_status(reply, StatusCode::NO_CONTENT))
 }
 
-#[actix_rt::main]
-async fn main() -> io::Result<()> {
+#[tokio::main]
+async fn main() {
     if get_env_or_default("JSON_LOGS", true) {
         use_json_log_subscriber()
     } else {
         use_unstructured_log_subscriber()
     }
 
-    let mut listenfd = ListenFd::from_env();
-
-    // Create Juniper schema
-    let schema = std::sync::Arc::new(create_schema());
-    let context = std::sync::Arc::new(create_context());
-
-    // Start http server
-    let mut server = HttpServer::new(move || {
-        App::new()
-            .data(schema.clone())
-            .data(context.clone())
-            .wrap(middleware::Logger::default())
-            .wrap(cors_middleware())
-            .service(web::resource("/graphql").route(web::post().to(graphql)))
-            .service(web::resource("/graphiql").route(web::get().to(graphiql)))
-            .service(web::resource("/healthz").route(web::get().to(healthz)))
-    });
-
-    server = if let Some(l) = listenfd.take_tcp_listener(0)? {
-        server.listen(l)?
-    } else {
-        server.bind(format!(
-            "0.0.0.0:{}",
-            std::env::var("PORT").unwrap_or_else(|e| {
-                eprintln!(
-                    "Error reading $PORT env var (defaulting to {}): {}",
-                    PORT, e
-                );
-                PORT.to_string()
-            })
-        ))?
-    };
-
-    server.run().await
+    let log = warp::log("medicines-api");
+
+    tracing::info!("hi");
+
+    let schema = create_schema();
+    let context = warp::any().map(create_context);
+
+    let cors = warp::cors()
+        .allow_methods(vec![Method::GET, Method::POST])
+        .allow_headers(vec![
+            header::AUTHORIZATION,
+            header::ACCEPT,
+            header::CONTENT_TYPE,
+        ])
+        .allow_any_origin();
+
+    let graphql_filter =
+        juniper_warp::make_graphql_filter(schema, context.boxed()).with(cors.clone());
+
+    let port = std::env::var("PORT").unwrap_or_else(|e| {
+        eprintln!(
+            "Error reading $PORT env var (defaulting to {}): {}",
+            PORT, e
+        );
+
+    warp::serve(
+        healthz()
+            .or(warp::path("graphiql").and(juniper_warp::graphiql_filter("/graphql", None)))
+            .or(warp::path("graphql").and(
+                warp::options()
+                    .map(warp::reply)
+                    .with(cors)
+                    .with(warp::log("cors-only")),
+            ))
+            .or(warp::path("graphql").and(graphql_filter))
+            .with(log),
+    )
+    .run(([127, 0, 0, 1], port))
```

Comment:
Does this work when running inside a container? Should this be `0.0.0.0` or something?